### PR TITLE
Feature/find entities by project

### PR DESCRIPTION
**Pull request by ChrisBerg**

### CODE REVIEW

File: src/main/java/nl/tudelft/thefirstorder/service/CameraService.java [hunk 1]
```diff
@@ -13,31 +13,39 @@
 
     /**
      * Save a camera.
-     * 
+     *
```
Comment (rubenwiersma) [inline src/main/java/nl/tudelft/thefirstorder/service/CameraService.java]: Empty rule is not really necessary, but not a major problem.

File: src/main/java/nl/tudelft/thefirstorder/service/impl/CameraServiceImpl.java [hunk 2]
```diff
@@ -55,16 +57,23 @@ public Camera save(Camera camera) {
...
+    @Transactional(readOnly = true)
     public Camera findOne(Long id) {
         log.debug("Request to get Camera : {}", id);
-        Camera camera = cameraRepository.findOne(id);
-        return camera;
+        return cameraRepository.findOne(id);
+    }
+
+    @Override
+    public List<Camera> findCamerasByProject(Long projectId) {
```
Comment (rubenwiersma) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/CameraServiceImpl.java]: Could this function not also be named findCameras(Long projectId)? By giving the projectId as attribute, you already imply that you're looking for a camera by project, it's not necessary to have it in the name of the function that way.

File: src/main/java/nl/tudelft/thefirstorder/service/PlayerService.java [hunk 3]
```diff
@@ -13,31 +13,39 @@
...
      */
     Page<Player> findAll(Pageable pageable);
 
     /**
+     * Get the players, associated with the map with the given id.
+     *
+     * @param projectId the id of the map.
+     * @return the list of player entities
+     */
+    List<Player> findPlayersByProject(Long projectId);
```
Comment (rubenwiersma) [inline src/main/java/nl/tudelft/thefirstorder/service/PlayerService.java]: Same naming issue as with findCameraByProject

**Comment by rubenwiersma** [comment]
Some conflicts that need to be resolved. Naming could possibly be shorter, but not a major issue.

File: src/main/java/nl/tudelft/thefirstorder/service/impl/CameraServiceImpl.java (hunk 2 above)
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/CameraServiceImpl.java]: Spring Data Repositories actually allow you to create queries based on the method name. So you might be able to delegate this method to `cameraRepository.findByProjectId(Long projectId)` and the implementation is magically created([see here](http://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html)). 
I agree that the name should be changed, but I would change it to the same name as the Repository, which is `findByProjectId`. The return type of Cameras is already implied by the service.

File: src/main/java/nl/tudelft/thefirstorder/service/impl/CueServiceImpl.java [hunk 4]
```diff
@@ -11,61 +14,82 @@
...
+    @Override
+    public List<Cue> findCuesByScript(Long scriptId) {
+        return StreamSupport
+            .stream(cueRepository.findAll().spliterator(), false)
+            .filter(cue -> Objects.equals(cue.getScript().getId(), scriptId))
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public Player getPlayer(Long cueId) {
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/CueServiceImpl.java]: A Cue object already has the Player and Camera in it. I think that querying just the Cue, also returns the associated Player and Camera.

File: src/main/java/nl/tudelft/thefirstorder/service/impl/MapServiceImpl.java [hunk 5]
```diff
@@ -47,72 +42,37 @@ public Map save(Map map) {
...
         mapRepository.delete(id);
     }
-
-    /**
-     * Adds a camera to the Map.
-     * @param mapId Id of the Map
-     * @param cameraId Id of the Camera
-     * @return The updated map
-     */
-    public Optional<Map> addCamera(Long mapId, Long cameraId) {
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/MapServiceImpl.java]: Why has this been removed?

File: src/main/java/nl/tudelft/thefirstorder/service/impl/MapServiceImpl.java [hunk 6]
```diff
@@ -47,72 +42,37 @@ public Map save(Map map) {
...
-
-    /**
-     * Adds a player to the Map.
-     *
-     * @param mapId    Id of the Map
-     * @param playerId Id of the Player
-     * @return The updated map
-     */
-    @Override
-    public Optional<Map> addPlayer(Long mapId, Long playerId) {
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/MapServiceImpl.java]: Same as above.

File: src/main/java/nl/tudelft/thefirstorder/service/impl/PlayerServiceImpl.java [hunk 7]
```diff
@@ -11,42 +11,52 @@
...
+    @Transactional(readOnly = true)
     public Page<Player> findAll(Pageable pageable) {
         log.debug("Request to get all Players");
-        Page<Player> result = playerRepository.findAll(pageable); 
-        return result;
+        return playerRepository.findAll(pageable);
+    }
+
+    @Override
+    public List<Player> findPlayersByProject(Long projectId) {
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/PlayerServiceImpl.java]: You might be able to use the Repository implementation. See my note at `CameraService.java`.

File: src/main/java/nl/tudelft/thefirstorder/service/util/XMLExportUtil.java [hunk 8]
```diff
@@ -1,175 +0,0 @@
-package nl.tudelft.thefirstorder.service.util;
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/service/util/XMLExportUtil.java]: Why did you remove this class?

File: src/main/java/nl/tudelft/thefirstorder/web/rest/MapResource.java [hunk 9]
```diff
@@ -143,40 +141,4 @@
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert("map", id.toString())).build();
     }
 
-    /**
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/web/rest/MapResource.java]: Why did you remove these methods?

File: src/main/java/nl/tudelft/thefirstorder/web/rest/ProjectResource.java [hunk 10]
```diff
@@ -184,39 +184,22 @@
     }
 
     /**
-     * GET /projects/:id/exportxml : exports the project to a XML.
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/web/rest/ProjectResource.java]: These shouldn't be removed?

File: src/main/resources/config/liquibase/changelog/20160504113134_added_entity_Cue.xml [hunk 11]
```diff
@@ -36,9 +36,6 @@
             <column name="camera_id" type="bigint">
                 <constraints nullable="true"/>
             </column>
-            <column name="project_id" type="bigint">
```
Comment (hungnguyen94) [inline src/main/resources/config/liquibase/changelog/20160504113134_added_entity_Cue.xml]: This shouldn't be changed. Your Cue table is now missing the `projectId` column.

File: src/main/resources/config/liquibase/changelog/20160504113134_added_entity_constraints_Cue.xml [hunk 12]
```diff
@@ -38,10 +38,5 @@
                                  referencedColumnNames="id"
                                  referencedTableName="camera"/>
 
-        <addForeignKeyConstraint baseColumnNames="project_id"
```
Comment (hungnguyen94) [inline src/main/resources/config/liquibase/changelog/20160504113134_added_entity_constraints_Cue.xml]: You should also not remove this.

File: src/main/resources/config/liquibase/changelog/20160524154401_added_jointable_map_cameras.xml [hunk 13]
```diff
@@ -1,18 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (hungnguyen94) [inline src/main/resources/config/liquibase/changelog/20160524154401_added_jointable_map_cameras.xml]: Don't remove this either.

File: src/main/resources/config/liquibase/changelog/20160524163221_added_jointable_map_players.xml [hunk 14]
```diff
@@ -1,18 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (hungnguyen94) [inline src/main/resources/config/liquibase/changelog/20160524163221_added_jointable_map_players.xml]: And this.

File: src/main/resources/config/liquibase/master.xml [hunk 15]
```diff
@@ -15,8 +15,6 @@
     <include file="classpath:config/liquibase/changelog/20160518111537_added_entity_TimePoint.xml" relativeToChangelogFile="false"/>
     <include file="classpath:config/liquibase/changelog/20160504134055_added_entity_CameraAction.xml" relativeToChangelogFile="false"/>
     <include file="classpath:config/liquibase/changelog/20160520223310_added_entity_column_User.xml" relativeToChangelogFile="false"/>
-    <include file="classpath:config/liquibase/changelog/20160524154401_added_jointable_map_cameras.xml" relativeToChangelogFile="false"/>
```
Comment (hungnguyen94) [inline src/main/resources/config/liquibase/master.xml]: Entries should not be removed.

File: src/main/webapp/app/entities/cue/cue-detail.controller.js [hunk 16]
```diff
@@ -5,9 +5,9 @@
         .module('thefirstorderApp')
         .controller('CueDetailController', CueDetailController);
 
-    CueDetailController.$inject = ['$scope', '$rootScope', '$stateParams', 'entity', 'Cue', 'Script', 'CameraAction', 'TimePoint', 'Player', 'Camera', 'Project'];
+    CueDetailController.$inject = ['$scope', '$rootScope', '$stateParams', 'entity', 'Cue', 'Script', 'CameraAction', 'TimePoint', 'Player', 'Camera'];
 
-    function CueDetailController($scope, $rootScope, $stateParams, entity, Cue, Script, CameraAction, TimePoint, Player, Camera, Project) {
```
Comment (hungnguyen94) [inline src/main/webapp/app/entities/cue/cue-detail.controller.js]: Cue does need the Project controller.

File: src/main/webapp/app/entities/cue/cue-detail.html [hunk 17]
```diff
@@ -28,11 +28,6 @@
             <a ui-sref="camera-detail({id:vm.cue.camera.id})">{{vm.cue.camera.name}}</a>
         </dd>
         <hr>
-        <dt><span>Project</span></dt>
```
Comment (hungnguyen94) [inline src/main/webapp/app/entities/cue/cue-detail.html]: Cue belongs to a project. Don't remove this.

File: src/main/webapp/index.html [hunk 18]
```diff
@@ -11,7 +11,7 @@
     <!-- bower:css -->
```
Comment (hungnguyen94) [inline src/main/webapp/index.html]: Don't commit `index.html`. Is not necessary.

File: src/test/java/nl/tudelft/thefirstorder/domain/AuthorityTest.java [hunk 19]
```diff
@@ -1,79 +0,0 @@
-package nl.tudelft.thefirstorder.domain;
```
Comment (hungnguyen94) [inline src/test/java/nl/tudelft/thefirstorder/domain/AuthorityTest.java]: **Removing** tests?!

File: src/main/java/nl/tudelft/thefirstorder/web/rest/CueResource.java [hunk 20]
```diff
@@ -97,10 +98,15 @@
         method = RequestMethod.GET,
         produces = MediaType.APPLICATION_JSON_VALUE)
     @Timed
-    public ResponseEntity<List<Cue>> getAllCues(Pageable pageable)
-        throws URISyntaxException {
+    public ResponseEntity<List<Cue>> getAllCues(Pageable pageable,
+                                                @RequestParam(required = false) Long scriptId)
+            throws URISyntaxException {
+        if(scriptId != null) {
```
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/web/rest/CueResource.java]: Checkstyle warning. `if` statements must be followed by whitespace.

**Comment by hungnguyen94** [comment]
A lot of classes, methods, tests and other things have been removed. This isn't mergeable until those have been restored.

File: src/main/java/nl/tudelft/thefirstorder/service/CameraService.java (hunk 1 above)
Comment (ChrisBerg) [inline src/main/java/nl/tudelft/thefirstorder/service/CameraService.java]: This javadoc format was used in all service Interfaces, so i added an extra space here as well ;)

File: src/main/java/nl/tudelft/thefirstorder/service/impl/MapServiceImpl.java (hunk 5 above)
Comment (ChrisBerg) [inline src/main/java/nl/tudelft/thefirstorder/service/impl/MapServiceImpl.java]: I have no idea, reintroduced it into the code.

**Comment by ChrisBerg** [comment]
I see, merging the dev into this branch has gone wrong somehow. Tomorrow i will fix that.
